client/txn.go

```diff
@@ -71,5 +174,28 @@ func (t *Txn) Prepare(calls ...Call) {
 func (t *Txn) Flush() error {
 	calls := t.prepared
 	t.prepared = nil
+	if len(calls) == 0 {
+		return nil
+	}
 	return t.kv.Run(calls...)
 }
+
+func (t *Txn) updateNeedsEndTxn(calls []Call) {
+	for _, c := range calls {
+		if b, ok := c.Args.(*proto.BatchRequest); ok {
```

Comment:
(because types don't match: the proto has RequestUnion as the type for the stuff in the batch, and BatchRequest isn't one of them).
Other clients like the python one might have to worry about this, but only on the client side.